src/librustc_mir/build/expr/as_constant.rs

```diff
@@ -45,6 +45,13 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                     literal,
                 }
             },
+            ExprKind::StaticRef { literal, .. } => {
```

Comment:
This variant should have a bit of documentation explaining that it is equivalent to `ExprKind::Constant´ except that it registers some diagnostic info

Reply:
I've added this to the doc comment on the variant.